### PR TITLE
click.open_file respects "errors" parameter.

### LINKED ISSUE #958: The `errors` argument in `click.open_file` is ignored on python3
The `errors` argument of `click.open_file` is not propagated to `sys.stdin` on python 3.x if there is no `encoding` argument in couple with `errors`.

How to reproduce:

```bash
> head -c64 /dev/urandom | python3 -c 'import click
print(click.open_file("-", errors="ignore").read())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.6/codecs.py", line 321, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcd in position 1: invalid continuation byte
```

It is because `_force_correct_text_reader` function checks that `_stream_is_misconfigured(text_reader)` call returns `False` and returns the `text_reader` as is.

### PR DESCRIPTION
**Pull request by joewledger**
Fixes https://github.com/pallets/click/issues/958

Signed-off-by: Joe Ledger <joewledger@gmail.com>

### CODE REVIEW

**Comment by joewledger** [comment]
Fixes https://github.com/pallets/click/issues/958

File: click/_compat.py [hunk 1]
```diff
@@ -482,9 +482,15 @@ def open_stream(filename, mode='r', encoding=None, errors='strict',
...
-        return io.open(filename, mode, encoding=encoding, errors=errors), True
+        open_kwargs = {}
+
+        if encoding is not None:
+            open_kwargs['encoding'] = encoding
+
+        if errors is not None:
+            open_kwargs['errors'] = errors
+
+        return codecs.open(filename, mode, **open_kwargs), True
```
Comment (davidism) [inline click/_compat.py]: Looks like `codecs.open` has an early return that ignores `errors`, so we need to continue using `io.open`. I'll add back your check for `"b" not in mode`.

File: click/_compat.py [hunk 2]
```diff
@@ -482,9 +482,15 @@ def open_stream(filename, mode='r', encoding=None, errors='strict',
 
     # Non-atomic writes directly go out through the regular open functions.
     if not atomic:
-        if encoding is None:
-            return open(filename, mode), True
-        return io.open(filename, mode, encoding=encoding, errors=errors), True
+        open_kwargs = {}
```
Comment (davidism) [inline click/_compat.py]: This needs to happen in atomic mode too, I'll move the kwargs outside the `if` block.

**Comment by davidism** [comment]
This is failing for `stdin` (`"-"`) because `_force_correct_text_reader` returns early if `encoding` isn't passed. I'll move those checks around.

**Comment by davidism** [comment]
Making the same changes to `_force_correct_text_writer` that I did for `reader` caused `test_termui.test_progressbar_update` to fail. For some reason it decided to write `\r` instead of `\n` newlines. Leaving a TODO in case someone works on that later.

**Comment by davidism** [comment]
A test is failing on Python 2 right now because `io.open(mode="w")` requires passing unicode to `write`. I'll see what I can do about that tomorrow.

**Comment by davidism** [comment]
I finally have this consistent. It applies to both read and write streams, makes standard and is consistent about using `io.open` and wrapping text mode on Python 2.

Thanks for adding the extra tests, they helped me ensure this was finished, I don't think I would have been nearly as thorough at refactoring it otherwise.

I was also able to identify why two tests were marked xfail for Windows Python 3: they were incorrectly writing `os.linesp` manually instead of `\n` with universal newline mode.

**Pull request merged**